core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -200,6 +201,39 @@ private static boolean skipItem(RexNode expr) {
           && "item".equalsIgnoreCase(((RexCall) expr).getOperator().getName());
   }
 
+  @Test void testGroupByDateLiteralSimple() {
+    HepProgramBuilder builder = new HepProgramBuilder();
+    builder.addRuleInstance(AggregateProjectConstantToDummyJoinRule.Config.DEFAULT.toRule());
+    HepPlanner hepPlanner = new HepPlanner(builder.build());
+
+    final String query = "select avg(sal)\n"
+        + "from emp\n"
+        + "group by DATE '2022-01-01'";
+    sql(query).withPlanner(hepPlanner).check();
```

Comment:
I think the test can be simplified by calling `withRule` instead of `withPlanner`.